enhancements/thindex.md

```diff
@@ -0,0 +1,113 @@
+---
+title: thindex-images
+authors:
+  - "@ecordell"
+reviewers:
+  - TBD
+approvers:
+  - TBD
+creation-date: 2020-04-25
+last-updated: 2020-04-25
+status: implementable
+---
+
+# Thindex Images
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+
+## Open Questions [optional]
+
+## Summary
+
+Operator Lifecycle Manager (OLM) currently talks to registry images (aka index images), containers which contain the registry database content and the grpc interface that serves data to consumers.
+
+This has the advantage of being very portable - index images can be moved from one cluster to another and easily mirrored.
+
+However, there is a major drawback. Because the server is embedded in the registry, care must be taken to ensure that the server is used with a compatible version of OLM (one that consumes the same version of the grpc api). 
+
+This enhancement describes a mechanism to decouple the server from the content being served, so that index images can be used forward-compatibly (in clusters with OLM installed at a newer version than the index was created for).
+
+This is the title of the enhancement. Keep it simple and descriptive. A good
+title can help communicate what the enhancement is and should be considered as
+part of any review.
+
+## Motivation
+
+Decouple the server components from the data served via index images today.
+
+## Proposal
+
+Define metadata that follows the [bundle image spec](https://github.com/openshift/enhancements/blob/master/enhancements/olm/operator-bundle.md).
+
+This matadata will be available both as labels on the final image, and inside the image in an `annotations.yaml` file:
```

Comment:
```suggestion
This metadata will be available both as labels on the final image and inside the image in an `annotations.yaml` file:
```

Reply:
Why does the metadata exist both as labels an a text file?

Reply:
That's because in an ideal world it would only exist as labels on the image, but certain kube distributions (ex. OpenShift) have limitations on being able to read image labels. At some point we will hopefully find a way around that issue and just trust the image labels.